textattack/search_methods/genetic_algorithm.py

```diff
@@ -19,184 +19,219 @@ class GeneticAlgorithm(SearchMethod):
+                pop_member.num_neighbors_list[rand_idx] = 0
+                break
 
-    def _crossover(self, pop_member1, pop_member2):
+            num_neighbors_list[rand_idx] = 0
+            iterations += 1
+
+    def _crossover(self, pop_member1, pop_member2, original_result):
         """
         Generates a crossover between pop_member1 and pop_member2.
+        If the child fails to satisfy the constraits, we re-try crossover for a fix number of times,
+        before taking one of the parents at random as the resulting child.
         Args:
-            pop_member1: The first population member.
-            pop_member2: The second population member.
+            pop_member1 (PopulationMember): The first population member.
+            pop_member2 (PopulationMember): The second population member.
         Returns:
             A population member containing the crossover.
         """
-        indices_to_replace = []
-        words_to_replace = []
         x1_text = pop_member1.attacked_text
         x2_words = pop_member2.attacked_text.words
-        new_neighbors_len = deepcopy(pop_member1.neighbors_len)
-        for i in range(len(x1_text.words)):
-            if np.random.uniform() < 0.5:
-                indices_to_replace.append(i)
-                words_to_replace.append(x2_words[i])
-                new_neighbors_len[i] = pop_member2.neighbors_len[i]
-        new_text = x1_text.replace_words_at_indices(
-            indices_to_replace, words_to_replace
-        )
-        return PopulationMember(new_text, deepcopy(new_neighbors_len))
 
-    def _get_neighbors_len(self, attacked_text):
+        num_tries = 0
+        passed_constraints = False
+        while num_tries < self.max_crossover_retries:
+            indices_to_replace = []
+            words_to_replace = []
+            num_neighbors_list = np.copy(pop_member1.num_neighbors_list)
+            for i in range(len(x1_text.words)):
+                if np.random.uniform() < 0.5:
+                    indices_to_replace.append(i)
+                    words_to_replace.append(x2_words[i])
+                    num_neighbors_list[i] = pop_member2.num_neighbors_list[i]
+            new_text = x1_text.replace_words_at_indices(
+                indices_to_replace, words_to_replace
+            )
+            new_text.attack_attrs["last_transformation"] = x1_text.attack_attrs[
+                "last_transformation"
+            ]
+            filtered = self.filter_transformations(
+                [new_text], x1_text, original_text=original_result.attacked_text
+            )
+
+            if filtered:
+                new_text = filtered[0]
+                passed_constraints = True
+                break
+
+            num_tries += 1
+
+        if not passed_constraints:
+            new_text = (
+                pop_member1.attacked_text
+                if np.random.uniform() < 0.5
+                else pop_member2.attacked_text
+            )
```

Comment:
I think this makes sense, but can you add a comment explaining what this is doing?